app/views/ReportingFrequencyView.scala.html

```diff
@@ -179,7 +179,7 @@ <h3 class="govuk-heading-s" id="rf-latency-small-heading">
     @if(viewModel.isAnyOfBusinessLatent) {
         @detailsDropdown("reporting.frequency.latency.details.section.header", latentBusinessText)
     }
-    @if(viewModel.displayManageYourRfSection) {
+    @if(viewModel.displayManageYourRfSection && viewModel.atLeastOneOfOptInOrOptOutExists) {
```

Comment:
Please move and this logic/rules to viewModel level, its hard to read and understand and maintain this code